### ISSUE TITLE
Fluent-bit stuck after receiving 500 error from Loki

### ISSUE THREAD

**Issue by kstinster** [fluent/fluent-bit#3969]
## Bug Report

**Describe the bug**
After receiving  500 error  from Loki, fluent-bit stops sending log towards Loki.
There are no attempts from fluent-bit to send logs to Loki, checked it with tcpdump
After restarting td-agent-bit service, fluent-bit  starts sending logs as normal. 

**To Reproduce**
We  can reproduce the problem by pushing high load to loki node, forcing to send 500 error. 

** Logs** 
At ~ 09:12 fluent-bit stopped sending logs, see  the log below at that time. 
```
Aug 17 09:12:05 logs01 td-agent-bit[10183]: [2021/08/17 09:12:05] [error] [output:loki:loki.0] loki-node01.ex.local:3100, HTTP status=500
Aug 17 09:12:05 logs01 td-agent-bit[10183]: rpc error: code = DeadlineExceeded desc = context deadline exceeded
Aug 17 09:12:05 logs01 td-agent-bit[10183]: [2021/08/17 09:12:05] [ warn] [engine] failed to flush chunk '10183-1629180715.248868234.flb', retry in 6 seconds: task_id=0, input=tail.0 > output=loki.0 (out_id=0)
Aug 17 09:12:16 logs01 td-agent-bit[10183]: [2021/08/17 09:12:16] [error] [output:loki:loki.0] loki-node01.ex.local:3100, HTTP status=400 Not retrying.
Aug 17 09:12:16 logs01 td-agent-bit[10183]: entry with timestamp 2021-08-17 06:11:58 +0000 UTC ignored, reason: 'entry out of order' for stream: {filename="/var/log/nginx/access.log", instance="nginx-node01",job="nginx-bit"}
Aug 17 09:12:16 logs01 td-agent-bit[10183]: total ignored: 1 out of 1
```
After several hours, we checked fluent-bit host, service is up,  counters were increasing .

```
Aug 17 12:56:41 logs01 td-agent-bit[10183]: ===== Input =====
Aug 17 12:56:41 logs01 td-agent-bit[10183]: tail.0 (tail)
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │
Aug 17 12:56:41 logs01 td-agent-bit[10183]: ├─ status
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │  └─ overlimit     : no
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │     ├─ mem size   : 192.7M (202011169 bytes)
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │     └─ mem limit  : 0b (0 bytes)
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │
Aug 17 12:56:41 logs01 td-agent-bit[10183]: ├─ tasks
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │  ├─ total tasks   : 2048
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │  ├─ new           : 2047
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │  ├─ running       : 1
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │  └─ size          : 133.2M (139633261 bytes)
Aug 17 12:56:41 logs01 td-agent-bit[10183]: │
Aug 17 12:56:41 logs01 td-agent-bit[10183]: └─ chunks
Aug 17 12:56:41 logs01 td-agent-bit[10183]:    └─ total chunks  : 2697
Aug 17 12:56:41 logs01 td-agent-bit[10183]:       ├─ up chunks  : 2697
Aug 17 12:56:41 logs01 td-agent-bit[10183]:       ├─ down chunks: 0
Aug 17 12:56:41 logs01 td-agent-bit[10183]:       └─ busy chunks: 2048
Aug 17 12:56:41 logs01 td-agent-bit[10183]:          ├─ size    : 133.2M (139633261 bytes)
Aug 17 12:56:41 logs01 td-agent-bit[10183]:          └─ size err: 0
Aug 17 12:56:41 logs01 td-agent-bit[10183]: ===== Storage Layer =====
Aug 17 12:56:41 logs01 td-agent-bit[10183]: total chunks     : 2697
Aug 17 12:56:41 logs01 td-agent-bit[10183]: ├─ mem chunks    : 2697
Aug 17 12:56:41 logs01 td-agent-bit[10183]: └─ fs chunks     : 0
Aug 17 12:56:41 logs01 td-agent-bit[10183]:    ├─ up         : 0
```

**Expected behavior**
fluent-bit  is expected to resend logs with some kind of backoff timers to Loki.

**Your Environment**
* Version used: 1.8.2, 1.8.1, 1.7.9
* Configuration:
```
[INPUT]
    Name        tail
    Tag         nginx-log
    Path        /var/log/nginx/*.log
    Path_Key          filename
    DB /opt/nginx.db
    Parser nginx_parser

[FILTER]
    Name geoip2
    Match *
    Database /etc/td-agent-bit/data/GeoLite2-City.mmdb
    Lookup_key remote_addr
    Record geoip_country_code remote_addr %{country.names.en}
    Record isocode remote_addr %{country.iso_code}

[OUTPUT]
    name loki
    Match *
    host loki-node01.ex.local
    port 3100
    labels job=nginx, env=prod,  instance=$hostname, host=$host

```

* Environment name and version :
* Server type and version: vmware vpshere vms , apt's and rpm's versions of fluent-bit
* Operating System and version: Ubuntu 18.04.5 LTS, CentOS Linux release 7.3.1611

**Additional context**
The issue  highly  affects our Environment, the problem happens occasionally and affects some of sources, so we need to find  sources with the problem and restart service.

**Comment by ohdearaugustin** [comment]
This problem sounds similar to #3328.

**Comment by kstinster** [comment]
We tried version up to 1.8.6,and still we experienced the issue. 
Then we replaced native loki output plugin on some nodes  with custom grafana-loki plugin https://grafana.com/docs/loki/latest/clients/fluentbit/.
After about two weeks of testing we have not had any issues on  nodes wtih grafana-loki , and at  the same times on nodes with native loki plugin the problem occurred several times. 
it seems like the issue is in native loki output plugin.
